pkgs/development/python-modules/pyarrow/default.nix

```diff
@@ -19,6 +19,7 @@ buildPythonPackage rec {
   PYARROW_BUILD_TYPE = "release";
 
   PYARROW_WITH_DATASET = true;
+  PYARROW_WITH_FLIGHT = true;
```

Comment:
We need to inherit this value from the enableFlight value passed to arrow-cpp

Reply:
Is that value accessible? I'm not aware of a way to access it

Reply:
You will need to add something like `passthru = { inherit enableFlight; }` to the expression for arrow-cpp